src/middlewared/middlewared/utils/asyncio_.py

```diff
@@ -1,10 +1,10 @@
 import asyncio
 
 
-async def asyncio_map(func, arguments, limit=None):
-    semaphore = None
-    if limit is not None:
-        semaphore = asyncio.BoundedSemaphore(limit)
+async def asyncio_map(func, arguments, limit=None, *, semaphore=None):
+    if limit is not None or semaphore is not None:
```

Comment:
Should we raise an exception if both limit and semaphore are specified as `limit` would be redundant in that case ?